osu.Game.Rulesets.Taiko/UI/DrumRollHitContainer.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Taiko.Objects.Drawables;
+using osu.Game.Rulesets.UI.Scrolling;
+
+namespace osu.Game.Rulesets.Taiko.UI
+{
+    internal class DrumRollHitContainer : ScrollingHitObjectContainer
+    {
+        protected override void Update()
+        {
+            base.Update();
+
+            // Remove any auxiliary hit notes that were spawned during a drum roll but subsequently rewound.
+            for (var i = AliveInternalChildren.Count - 1; i >= 0; i--)
+            {
+                var flyingHit = (DrawableFlyingHit)AliveInternalChildren[i];
+                if (Time.Current < flyingHit.HitObject.StartTime)
```

Comment:
```suggestion
                if (Time.Current <= flyingHit.HitObject.StartTime)
```

Reply:
are we guaranteed to always be > starttime at the point of display?

Reply:
Don't see why we would. This is covering for the case of rewinding exactly onto the hit time, which is a pretty edge case scenario for sure, but doesn't guarantee > starttime.

Reply:
not only is this wrong, both are wrong. can easily see why in test scene. the hitobject (tick) start time here isn't the right time to use, it's the judgement time that matter. even the original code is wrong.